### PR TITLE
fix: ensure migrated pages are added to config.matcher in middleware

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- For app router pages, we set custom headers in `middleware` which are always used in generating metadata (e.g., `x-locale` header that contains the user's locale)
- The middleware is just additional code running within the same function invocation - it doesn't trigger separate/additional serverless function calls that would increase costs.
- This PR will fix this issue where the metadata is not translated for some migrated pages

Screenshot 2024-12-12 at 11 03 14 PM

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/middleware.ts [hunk 1]
```diff
@@ -153,6 +153,13 @@ export const config = {
   // Next.js Doesn't support spread operator in config matcher, so, we must list all paths explicitly here.
   // https://github.com/vercel/next.js/discussions/42458
   matcher: [
+    "/d/:path*",
+    "/more/:path*",
+    "/maintenance/:path*",
+    "/enterprise/:path*",
+    "/upgrade/:path*",
+    "/connect-and-join/:path*",
+    "/insights/:path*",
```
Comment (hbjORbj) [inline apps/web/middleware.ts:162]: all of these routes are already migrated to app router in prod

Comment (hariombalhara) [reply apps/web/middleware.ts:162]: Would these mean that in the end all routes go through middleware? Wouldn't it increase function invocations cost?

Comment (hbjORbj) [reply apps/web/middleware.ts:162]: > Would these mean that in the end all routes go through middleware? 

Yes. The design of app router is this way, we set customer headers and use them extensively

Comment (hbjORbj) [reply apps/web/middleware.ts:162]: > Wouldn't it increase function invocations cost a ton?
The middleware runs in the same execution context as your route handler - it's not spawning new serverless functions

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**